flamingpy/utils/viz.py

```diff
@@ -284,6 +305,139 @@ def draw_EGraph(
+            the tuple should be of the form:
+
+                ``("weight", {-1: minus_color, +1: plus_color})``
+
+        label (NoneType or string): plot values next to each node
+            associated with the node attribute label. For example,
+            to plot bit values, set label to "bit_val". If set to 'index',
+            it will plot the integer indices of the nodes. If the attribute
+            for some or all of the nodes, a message will print indicating
+            for how many nodes the attribute has not been set.
+        title (bool): if True, display the title, depending on the label.
+            For default labels, the titles are converted from attribute
+            name to plane English and capitalized.
+        legend (bool): if True and color_nodes argument is a tuple(str, dict),
+            display the a color legend with node attributes.
+        show_axes (bool): if False, turn off the axes.
+        dimensions (tuple): Dimensions of the region that should be plotted.
+            Should be of the form:
+
+                ``([xmin, xmax], [ymin, ymax], [zmin, zmax])``
+
+            If None, sets the dimensions to the smallest rectangular space
+            containing all the nodes.
+
+    Returns:
+        figure: Plotly Express figure.
+    """
+
+    nodes = np.array(egraph.nodes)
+    x_nodes, y_nodes, z_nodes = nodes[:, 0], nodes[:, 1], nodes[:, 2]
+
+    x_edges = []
+    y_edges = []
+    z_edges = []
+
+    for edge in egraph.edges:
+        x0, y0, z0 = edge[0]
+        x1, y1, z1 = edge[1]
+        x_edges.extend([x0, x1, None])
+        y_edges.extend([y0, y1, None])
+        z_edges.extend([z0, z1, None])
+
+    nodeColors = []
+    for point in egraph.nodes:
+        color = _get_node_color(egraph, color_nodes, point)
+        nodeColors.append(color)
+
+    # nodes
+    node_trace = go.Scatter3d(
+        name="nodes",
+        x=x_nodes,
+        y=y_nodes,
+        z=z_nodes,
+        mode="markers",
+        marker=dict(
+            symbol="circle",
+            size=5,
+            color=nodeColors,
+            line=dict(color="black", width=0.5),
+        ),
+        hoverinfo="none",
+    )
+
+    # edges
+    edge_trace = go.Scatter3d(
+        name="edges",
+        x=x_edges,
+        y=y_edges,
+        z=z_edges,
+        line=dict(color="black", width=1),
+        mode="lines",
+        marker=dict(color="black"),
+        hoverinfo="none",
+    )
+
+    axis = dict(
+        showbackground=True,
+        showline=True,
+        zeroline=False,
+        showgrid=True,
+        showticklabels=True,
+        tickmode="linear",
+        tick0=0,
+        dtick=1,
+    )
+
+    layout = go.Layout(
+        width=750,
+        height=750,
+        showlegend=legend,
+        hovermode="closest",
+        scene=dict(
+            xaxis={**dict(title="x"), **axis},
+            yaxis={**dict(title="y"), **axis},
+            zaxis={**dict(title="z"), **axis},
```

Comment:
I'm not sure if this is according to our weird convention 🙈 @ilan-tz ?

Reply:
The x, y, z convention is flipped for the plots. On the matplotlib side we scatter points and connect edges with (x, z, y) coords instead of (x, y, z).

Reply:
This is fine as long as the axes are labelled correctly